### PR TITLE
Update preprocessing for milestone 2

### PR DESCRIPTION
**Pull request by cmmclaug**
* Updated the preprocessing script to clean the data as discussed, and split into train/test/deploy datasets.
* The year column is now a numeric year, the first_appearances is now a datetime.

### CODE REVIEW

**Review by aidanmattrick — APPROVED** [review]
Looks good to me!
Good  call on creating separate DF to store observations for which we don't have alignment info on.

File: src/preprocess_data.py [hunk 1]
```diff
@@ -54,18 +77,34 @@ def preprocess_data():
...
     if verbose: print("\nOutput data summary:")
     if verbose: print(f"{characters_data.info()}")
 
+    # Creating deployment data file from rows missing target values")
+    deploy_df = characters_data[characters_data['align'].isnull()]
+    deploy_df.to_csv(output_file_slug + "_deploy." + output_file_ext)
+    characters_data = characters_data[characters_data['align'].notna()]
+
+    # Split train and test data
+    train_df, test_df = train_test_split(characters_data, test_size=0.3, random_state=123)
```
Comment (zmerpez) [inline src/preprocess_data.py]: For the test_size=0.3, this is a bit higher, we may choose test_size=0.2 or 0.25?

**Review by zmerpez — COMMENTED** [review]
I just have one concern about test_size. If we got good enough train set size, thats fine, else we may change with 0.2 or 0.25?

File: src/preprocess_data.py (hunk 1 above)
Comment (zmerpez) [reply src/preprocess_data.py]: Also, in align we have just 3 rows with value as "Reformed", are we deleting those as well ?

Comment (cmmclaug) [reply src/preprocess_data.py]: I will add a line to change reformed to "Good" in our cleaned data, and lower the test size to 0.2.

**Review by zmerpez — APPROVED** [review]

**Pull request merged**